src/i18_bluesky/plans/grid_scan_diode.py

```diff
@@ -0,0 +1,28 @@
+import bluesky.plans as bp
+from dodal.common import MsgGenerator, inject
+from dodal.devices.i18.diode import Diode
+from dodal.devices.i18.table import Table
+from dodal.plans.data_session_metadata import attach_data_session_metadata_decorator
+
+DIODE = inject("diode")
+
+TABLE = inject("table")
+
+
+# todo make it use scanspec instead
+
+
+@attach_data_session_metadata_decorator()
+def grid_scan_diode(
+    x_start: float,
+    x_stop: float,
+    x_steps: int,
+    y_start: float,
+    y_stop: float,
+    y_steps: int,
+    diode: Diode = DIODE,
+    table: Table = TABLE,
+) -> MsgGenerator:
```

Comment:
hmm, 
alternatively we could write a manual scan name for those use cases, to be used with blueapi

not sure

@callumforrester what is the recommended way of doing this?

Reply:
The recommended way is to write a generic plan for a grid, then have a specific form with the device that is to be used as a default.

Reply:
@iain-hall what can we say that would be the default device? the stage?

Reply:
thanks!

>  other types of scan as well (e.g. pinhole scan measuring on ionchambers). 


we don't have this in the soon issues, can we merge this and consider the pinhole scan once it comes up on the list? then we can make it generic should we need to

Reply:
@stan-dot But having a plan to scan the diode that involves the table makes no sense! A plan to scan the table with a list of detectors would be useful though. So can we just rename the PR to 'simple plan to scan the table' and change the name of the plan to 'grid_scan_table' and use a list of detectors rather than the diode?
BTW we already have a generic plan for grid scans in src/i18_bluesky/plans/simple_plans.py that tables arbitrary x, y motors and detectors.

Reply:
ok we already have a scan plan for this, there is no need for this PR